### PR TITLE
Convert string tensors to python strings in `generate()`

### PR DESCRIPTION
**Pull request by mattdangerw**
There may be more to discuss on whether we put this in a preprocessor layer eventually, but for now just updating our `generate()` function to return python lists of python strings when preprocessing is attached.

### CODE REVIEW

**Review by chenmoneygithub — APPROVED** [review]
Thanks!

File: keras_nlp/models/gpt2/gpt2_causal_lm.py [hunk 1]
```diff
@@ -437,4 +438,8 @@ def preprocess(x):
             outputs.append(output)
 
         outputs = tf.concat(outputs, axis=0)
-        return tf.squeeze(outputs, 0) if input_is_scalar else outputs
+        outputs = tf.squeeze(outputs, 0) if input_is_scalar else outputs
```
Comment (jbischof) [inline keras_nlp/models/gpt2/gpt2_causal_lm.py:441]: Where does this check if `preprocessor` is `None`? Are we just relying on the output being a `tf.string`?

**Review by jbischof — APPROVED** [review]
Thanks!

File: keras_nlp/models/gpt2/gpt2_causal_lm.py (hunk 1 above)
Comment (jbischof) [reply keras_nlp/models/gpt2/gpt2_causal_lm.py:441]: Also, to be fully consistent with `predict` would we need to return token outputs as `ints` as well?

Comment (mattdangerw) [reply keras_nlp/models/gpt2/gpt2_causal_lm.py:441]: To be fully consistent with `predict` would not be `ints` I think, but numpy arrays? @fchollet was saying that numpy string arrays are both inefficient and hard to work with, but in the case of numerical return types we definitely return numpy. Maybe the thing to do is this...

```
# Convert outputs to a friendly pythonic type.
# For numerical outputs that is numpy, for string outputs that is `list` and `str`.
if outputs.dtype == tf.string:
    return tensor_to_string_list(outputs)
else:
    return outputs.numpy()
```

Comment (mattdangerw) [reply keras_nlp/models/gpt2/gpt2_causal_lm.py:441]: Re the check. `if outputs.dtype == tf.string:` and `if self.preprocessor is None:` are equivalent in all cases. Just depends on how you want to think about it.

Comment (jbischof) [reply keras_nlp/models/gpt2/gpt2_causal_lm.py:441]: OK, this new version seems more consistent. Thanks!

**Review by jbischof — APPROVED** [review]
Looks great!

**Pull request merged**